typescript/cli/src/tests/ethereum/submit/submit.e2e-test.ts

```diff
@@ -155,4 +156,86 @@ describe('hyperlane submit', function () {
+      // Generate a random-ish filename for local testing because FileSubmitterStrategy always appends so tests may fail
+      const outputTransactionPath = `${TEMP_PATH}/transactions_${randomInt(0, 1_000_000)}.json`;
+      const fileSubmitterStrategy = {
+        [CHAIN_NAME_2]: {
+          submitter: {
+            type: CustomTxSubmitterType.FILE,
+            filepath: outputTransactionPath,
+          },
+        },
+        [CHAIN_NAME_3]: {
+          submitter: {
+            type: CustomTxSubmitterType.FILE,
+            filepath: outputTransactionPath,
+          },
+        },
+      };
+
+      const strategyPath = `${TEMP_PATH}/file-strategy.yaml`;
+      writeYamlOrJson(strategyPath, fileSubmitterStrategy);
+
+      const chain2MintAmount = randomInt(1, 1000);
+      const chain3MintAmount = randomInt(1, 1000);
+      const transactions = await Promise.all([
+        getMintOnlyOwnerTransaction(
+          xerc20Chain2,
+          ALICE,
+          chain2MintAmount,
+          ANVIL2_CHAIN_ID,
+        ),
+        getMintOnlyOwnerTransaction(
+          xerc20Chain3,
+          BOB,
+          chain3MintAmount,
+          ANVIL3_CHAIN_ID,
+        ),
+      ]);
+      const transactionsPath = `${TEMP_PATH}/strategy-test-transactions.yaml`;
+      writeYamlOrJson(transactionsPath, transactions);
+
+      await hyperlaneSubmit({ strategyPath, transactionsPath });
+
+      const outputtedTransactions = readYamlOrJson(outputTransactionPath);
+      expect(outputtedTransactions).to.have.deep.members(transactions); // members is an array of values that are unordered
+    });
+
+    it('should overwrite a transactions file if it is malformed (not array)', async () => {
+      const outputTransactionPath = `${TEMP_PATH}/transactions_${randomInt(0, 1_000_000)}.json`;
+      const fileSubmitterStrategy = {
+        [CHAIN_NAME_2]: {
+          submitter: {
+            type: CustomTxSubmitterType.FILE,
+            filepath: outputTransactionPath,
+          },
+        },
+      };
+
+      const strategyPath = `${TEMP_PATH}/file-strategy.yaml`;
+      writeYamlOrJson(strategyPath, fileSubmitterStrategy);
+
+      // Add an invalid transaction object (we expect arrays)
+      const transactionsPath = `${TEMP_PATH}/strategy-test-transactions.yaml`;
+      writeYamlOrJson(transactionsPath, { invalid: 'transaction' });
+
+      const chain2MintAmount = randomInt(1, 1000);
+      const transactions = await Promise.all([
+        getMintOnlyOwnerTransaction(
+          xerc20Chain2,
+          ALICE,
+          chain2MintAmount,
+          ANVIL2_CHAIN_ID,
+        ),
+      ]);
+      writeYamlOrJson(transactionsPath, transactions);
+
+      await hyperlaneSubmit({ strategyPath, transactionsPath });
+      const outputtedTransactions = readYamlOrJson(outputTransactionPath);
+      expect(outputtedTransactions).to.have.deep.members(transactions); // members is an array of values that are unordered
+    });
```

Comment:
Can we add an assertion for checking that the original data is not in the final file?